guides/hack/11-built-in-types/35-enum-class.md

```diff
@@ -169,6 +169,33 @@ If either `D1`, `D2` or `D3` tries to define a constant named `Age`, there will
 
 Though the `final` keyword is not supported, Enum classes support the [`__Sealed`](../attributes/predefined-attributes#__sealed) attribute. Using `__Sealed`, you can specify which other enum classes, if any, are allowed to extend from your enum class.
 
+
+## Abstract enum classes
+
+Like regular classes, enum classes come in two flavors: a concrete one and an abstract one. An abstract enum class can declare abstract members, where only their type and name are provided.
+
+```EnumClassIntro.Abstract.hack no-auto-output
+// abstract enum class with some abstract members
+abstract enum class AbstractNames : IHasName {
+  abstract HasName Foo;
+  HasName Bar = new HasName('bar');
+}
+```
+
+Abstract members do not support default values, and can't be accessed directly. They only map a name to a type.
+One must extends abstract enum classes into concrete ones, and provide all the abstract members to build a valid enum class.
```

Comment:
```suggestion
For an enum class with abstract members to successfully build and typecheck, you must extend your [abstract class](/hack/classes/inheritance#abstract-classes) with implementations of all abstract members.
```
Switching to active voice and rewording because "One must extends abstract enum classes" felt a bit off. If any of my other suggestions are not strictly correct, feel free to ignore.

Reply:
Ok, I'll rephrase this. But you're confusing classes and enum classes. I'll ping you once this is updated.